contracts/src/steel/OpSteel.sol

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.9;
+
+import {Encoding, Steel} from "./Steel.sol";
+
+abstract contract OpCommitmentValidator {
+    /// @notice Address of the OptimismPortal2 contract.
+    IOptimismPortal2 public immutable optimismPortal;
+
+    constructor(address _optimismPortal) {
+        optimismPortal = IOptimismPortal2(_optimismPortal);
+    }
+
+    /// @notice Validates a Steel commitment.
+    /// @param commitment The commitment to validate.
+    /// @return True if the commitment is valid, false otherwise.
+    function validateCommitment(Steel.Commitment calldata commitment) internal view returns (bool) {
+        (uint240 blockID, uint16 version) = Encoding.decodeVersionedID(commitment.id);
+        if (version == 0x100) {
+            return validateDisputeGameCommitment(blockID, commitment.digest);
+        } else {
+            return Steel.validateCommitment(commitment);
+        }
+    }
+
+    /// @notice Validates a Dispute Game commitment.
+    /// @param gameIndex The index of the game in the DisputeGameFactory.
+    /// @param rootClaim The root claim of the dispute game.
+    /// @return True if the commitment is valid, false otherwise.
+    function validateDisputeGameCommitment(uint256 gameIndex, bytes32 rootClaim) internal view returns (bool) {
+        IDisputeGameFactory factory = optimismPortal.disputeGameFactory();
+
+        // Retrieve game information from the factory.
+        (uint32 gameType, uint64 createdAt, IDisputeGame game) = factory.gameAtIndex(gameIndex);
+
+        // The game type of the dispute game must be the respected game type.
+        if (gameType != optimismPortal.respectedGameType()) return false;
+        // The game must have been created after `respectedGameTypeUpdatedAt`.
+        if (createdAt < optimismPortal.respectedGameTypeUpdatedAt()) return false;
+        // The game must be resolved in favor of the root claim (the output proposal).
+        if (game.status() != GameStatus.DEFENDER_WINS) return false;
+        // The game must have been resolved for at least `proofMaturityDelaySeconds`.
+        if (block.timestamp - game.resolvedAt() <= optimismPortal.proofMaturityDelaySeconds()) return false;
+        // The game must not be blacklisted.
+        if (optimismPortal.disputeGameBlacklist(game)) return false;
```

Comment:
I don't like the OP design here because it makes `validateDisputeGameCommitment` non-monotonic:
- `disputeGameBlacklist` only returns a boolean, not a timestamp
- If the guardian blacklists a game long after its maturity delay at time $ts$, `validateDisputeGameCommitment` will succeed before $ts$, but then suddenly fail when rerunning on the same input after $ts$

I don't see a good fix for this. I guess removing the `disputeGameBlacklist` check would even lead to more inconsistencies.